data/calculated_parser/functions.py

```diff
@@ -408,6 +410,106 @@ def depthexcess(depth, latitude, temperature, salinity, bathy) -> np.ndarray:
+    channel_end_depth = np.empty_like(Cmax,dtype='float')
+    it = np.nditer(Cmax,flags=['refs_ok','multi_index'])
+    for x in it:
+        channel_end_depth[it.multi_index] = np.interp(x, soundspeed[:,it.multi_index[0], it.multi_index[1]],depth)
+    del_Z = channel_end_depth-channel_start_depth
+    numerator = freq_cutoff * del_Z
+    denominator = 0.2652 * Cmin
+    final_denom = numerator/denominator 
+    final_denom = np.power(final_denom,2)
+    delC = Cmin/final_denom
+    #print(delC)
+    return delC
+
+def potentialsubsurfacechannel(depth, latitude,temperature, salinity, freq_cutoff = 2755.03 )-> np.ndarray:  
+    """
+     Detect if there is sub-surface channel. 
+     Required Arguments:
+        * depth: Depth in meters
+        * latitude: Latitude in degrees North
+        * temperature: Temperatures in Celsius
+        * salinity: Salinity
+        * freq_cutoff: Desired frequency cutoff in Hz
+     Returns 1 if the profile has a sub-surface channel, 0 if the profile does not have a sub-surface channel
+    """
+    depth, latitude, temperature, salinity = __validate_depth_lat_temp_sal(
+        depth, latitude, temperature, salinity)
+
+    # Trimming the profile considering the depth above 1000m
+    depth = depth[depth<1000]
+    depth_length = len(depth)
+    temp = temperature[0:depth_length,:,:]
+    sal =  salinity[0:depth_length,:,:]
+    sound_speed = sspeed(depth, latitude, temp, sal)
+    minima = np.apply_along_axis(spsignal.find_peaks,0,-sound_speed)[0] 
+    maxima = np.apply_along_axis(spsignal.find_peaks,0, sound_speed)[0]
+    delC = calculate_del_C(depth,sound_speed,minima,maxima, freq_cutoff)
+    hasPSSC = np.empty_like(minima, dtype='float')
+    
+    it = np.nditer(minima,flags=['refs_ok','multi_index'])
+    for minima_array in it:
+        minima_list = minima_array.tolist()
+        maxima_list = maxima[it.multi_index].tolist()
+        if len(minima_list)>=2:
+            p1 = 0
+            p2 = minima[it.multi_index].tolist()[0]
+            if len(maxima_list) >=2:
+                p1 = maxima_list[0]
+                p3 = maxima_list[1]
+            else:
+                p3= maxima_list[0]
+            if p3 < p2: #if the only maximum is higher in the water column than the minima
+                hasPSSC[it.multi_index]=0
+            else:
+                p1_sound_speed = sound_speed[p1,it.multi_index[0], it.multi_index[1]]
+                p2_sound_speed = sound_speed[p2,it.multi_index[0], it.multi_index[1]]
+                p3_sound_speed = sound_speed[p3,it.multi_index[0], it.multi_index[1]]
+                
+                c1 = abs(p1_sound_speed-p2_sound_speed) 
+                c2 = abs(p3_sound_speed-p2_sound_speed)
+                
+                if c1> delC[it.multi_index] and c2> delC[it.multi_index]: 
+                    hasPSSC[it.multi_index]= 1
+                else:
+                    hasPSSC[it.multi_index]= 0
+        else:
+            hasPSSC[it.multi_index] = 0
```

Comment:
- Not because of the `multi_index` I believe. As I use the  _nditer_ for iterating over the _ndarray_ which contains 1D ndarray as each of its element when I try to get the length of the ndarray like below - 
![nditer snapshot](https://user-images.githubusercontent.com/12777798/116248700-a738d100-a746-11eb-8397-71d816c23df0.png)
I tried _shape, size and len_  on the element. _shape_ returned an empty tuple, _size_ returned 1 for all of them, _len()_ displayed an error `len() of unsized object`. The only thing that worked is converting it to a list and check the size after that. Is there a work around for this?

Reply:
I have changed the part of the code where I use `np.empty_like()`, and made it use 'np.zeros_like()`. It removed several `else` paths on the code and will certainly be faster. :smile:

Reply:
I have replaced the use of `np.empty_like()`s to `np.zeros_like()`. It removed several `else` paths from the function and made it faster. :smile: